flutter/ios/Classes/SentryFlutterPluginApple.swift

```diff
@@ -258,65 +258,21 @@ public class SentryFlutterPluginApple: NSObject, FlutterPlugin {
     private func captureEnvelope(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
         guard let arguments = call.arguments as? [Any],
               !arguments.isEmpty,
-              let event = arguments.first as? String else {
+              let data = (arguments.first as? FlutterStandardTypedData)?.data  else {
             print("Envelope is null or empty !")
             result(FlutterError(code: "2", message: "Envelope is null or empty", details: nil))
             return
         }
-
-        do {
-            let envelope = try parseJsonEnvelope(event)
-            SentrySDK.capture(envelope)
-            result("")
-        } catch {
-            print("Cannot parse the envelope json !")
-            result(FlutterError(code: "3", message: "Cannot parse the envelope json", details: nil))
+        // We need to replace this when we have the possibility to access envelope deserialization.
+        guard let sentrySerialization = NSClassFromString("SentrySerialization") as? NSObject.Type,
+            let unmanaged = sentrySerialization.perform(NSSelectorFromString("envelopeWithData:"), with: data),
+            let envelope = unmanaged.takeUnretainedValue() as? SentryEnvelope else {
+            print("Cannot parse the envelope data")
+            result(FlutterError(code: "3", message: "Cannot parse the envelope data", details: nil))
```

Comment:
@bruno-garcia @philipphofmann can we figure this out before merging?
using unmanaged code is too flaky

Reply:
Do you mean you want to be able to access `SentrySerialization.envelopeWithData`? If yes, we could add it there: https://github.com/getsentry/sentry-cocoa/issues/1120

Reply:
yes, that's the case, just don't remember why the private header didn't work, @denrase ?

Reply:
I just opened a PR for making this public https://github.com/getsentry/sentry-cocoa/pull/1131.

Reply:
Just did a release for that https://github.com/getsentry/sentry-cocoa/releases/tag/7.1.3. You can now using `PrivateSentrySDKOnly`  @denrase.